packages/filecoin-api/src/storefront/events.js

```diff
@@ -34,7 +38,28 @@ export const handleFilecoinSubmitMessage = async (context, message) => {
     }
   }
 
-  // TODO: verify piece
+  // read and compute piece for content
+  // TODO: needs to be hooked with location claims
+  const contentGetRes = await context.dataStore.get(message.content)
```

Comment:
here we actually need to decide this interface. Being a datastore would be nice to allow implementers to use R2/S3 client kind of things, which would decrease costs and enhance performance. However, probably we do not want the datastore to be hooked with the claims itself, so probably datastore should in the future receive claims for the content?

Reply:
Not sure I understand - the datastore can use content claims under the hood no?

Reply:
`.get()` implies to me it's retrieving the whole data, perhaps consider calling this `.stream()`? (similar to https://developer.mozilla.org/en-US/docs/Web/API/Blob/stream)

Reply:
yeah good point, I will change to stream. 

> Not sure I understand - the datastore can use content claims under the hood no?

yes it can. But probably we would like it to be within the protocol implementation rather than next to the infra wiring? Let's discuss it when we get there